iree/compiler/Conversion/LinalgToLLVM/HALInterfaceToMemrefArguments.cpp

```diff
@@ -122,6 +127,11 @@ struct ProcessFuncInterfacePattern : public OpConversionPattern<FuncOp> {
         ++argIndex;
       }
     }
+    SmallVector<int64_t, 4> shapes;
```

Comment:
Is this `shapes` variable necessary? I think we can just do `auto dynamicDimsBufferType = MemRefType::get(ShapedType::kDynamicSize, rewriter.getIntegerType(32))`?